src/ga4gh/vrs/normalize.py

```diff
@@ -3,60 +3,192 @@
-                                           alleles=alleles,
-                                           mode=NormalizationMode.EXPAND,
-                                           anchor_length=0)
+        trim_ival, trim_alleles = _normalize(ref_seq, ival, alleles, mode=None, trim=True)
+    except ValueError:
+        # Occurs for ref agree Alleles (when alt = ref)
+        len_ref_seq = len_alt_seq = 0
+    else:
+        trim_ref_seq = ref_seq[trim_ival[0]: trim_ival[1]]
+        trim_alt_seq = trim_alleles[1]
+        len_ref_seq = len(trim_ref_seq)
+        len_alt_seq = len(trim_alt_seq)
 
-        new_allele.location.start = new_ival[0]
-        new_allele.location.end = new_ival[1]
+    # Compare the two allele sequences
+    if not len_ref_seq and not len_alt_seq:
+        return input_allele
 
-        if new_allele.state.type in _states_with_sequence:
-            new_allele.state.sequence = models.SequenceString(new_alleles[1])
+    new_allele = pydantic_copy(allele)
+
+    if len_ref_seq and len_alt_seq:
+        new_allele.location.start = _get_new_allele_location_pos(
+            trim_ival[0], start.pos_type
+        )
+        new_allele.location.end = _get_new_allele_location_pos(
+            trim_ival[1], end.pos_type
+        )
+        new_allele.state.sequence = models.SequenceString(trim_alleles[1])
+        if sequence_reference:
+            new_allele.location.sequence = sequence_reference
+        return new_allele
+
+    # Determine bounds of ambiguity
+    try:
+        new_ival, new_alleles = _normalize(
+            ref_seq,
+            trim_ival,
+            (None, trim_alleles[1]),
+            mode=NormalizationMode.EXPAND
+        )
     except ValueError:
         # Occurs for ref agree Alleles (when alt = ref)
         pass
+    else:
+        new_allele.location.start = _get_new_allele_location_pos(
+            new_ival[0], start.pos_type
+        )
+        new_allele.location.end = _get_new_allele_location_pos(
+            new_ival[1], end.pos_type
+        )
+
+        new_ref_seq = ref_seq[new_ival[0]: new_ival[1]]
+
+        if not new_ref_seq:
+            # If the reference sequence is empty this is an unambiguous insertion.
+            # Return a new Allele with the trimmed alternate sequence as a Literal
+            # Sequence Expression
+            new_allele.state = models.LiteralSequenceExpression(
+                sequence=models.SequenceString(new_alleles[1])
+            )
+        else:
+            # Otherwise, return a new Allele using a RLE
+            sequence = models.SequenceString(new_alleles[1])
+            len_sequence = len(sequence.root)
+
+            new_allele.state = models.ReferenceLengthExpression(
+                length=len_sequence,
+                repeatSubunitLength=len_ref_seq or len_alt_seq
+            )
+
+            if (rle_seq_limit and len_sequence < rle_seq_limit) or (rle_seq_limit is None):
+                new_allele.state.sequence = sequence
```

Comment:
Very minor change that eliminates string duplication operation when `len_sequence >= rle_seq_limit`
```suggestion
            len_sequence = len(new_alleles[1])

            new_allele.state = models.ReferenceLengthExpression(
                length=len_sequence,
                repeatSubunitLength=len_ref_seq or len_alt_seq
            )

            if (rle_seq_limit and len_sequence <= rle_seq_limit) or (rle_seq_limit is None):
                new_allele.state.sequence = models.SequenceString(new_alleles[1])
```